compiler/src/iree/compiler/Dialect/Flow/IR/FlowOps.cpp

```diff
@@ -495,15 +495,15 @@ LogicalResult DispatchTieShapeOp::verify() {
 
 LogicalResult DispatchTieShapeOp::reifyResultShapes(
     OpBuilder &b, ReifiedRankedShapedTypeDims &reifiedReturnShapes) {
-  SmallVector<Value> shape;
+  SmallVector<OpFoldResult> shape;
   unsigned dynamicIdx = 0;
   auto tensorType =
       getResult().getType().cast<IREE::Flow::DispatchTensorType>();
   for (int64_t dim : tensorType.getShape()) {
     if (dim == ShapedType::kDynamic) {
-      shape.push_back(getDynamicDims()[dynamicIdx++]);
+      shape.push_back((Value)getDynamicDims()[dynamicIdx++]);
```

Comment:
Can we avoid the casting like this... it's a red flag to me. My intuition tells me that `DynamicDims` is either `Value` type or `OpFoldResult` type. Do we really need a cast here?

Reply:
You can't create an `OpFoldResult` without casting. DynamicsDims are always `Value`. You can try it out - it will not compile.